grails-app/domain/org/pih/warehouse/invoice/InvoiceItem.groovy

```diff
@@ -82,8 +82,17 @@ class InvoiceItem implements Serializable {
         product(nullable: true)
         glAccount(nullable: true)
         budgetCode(nullable: true)
-
-        quantity(nullable: false, min: 0) // min = 0 for canceled items
+        quantity(nullable: false, min: 0, validator: { quantity, obj ->
+            InvoiceItem.withNewSession {
+                InvoiceItem notUpdatedInvoiceItem = InvoiceItem.get(obj?.id)
+                if (notUpdatedInvoiceItem) {
+                    Boolean isValid = quantity + (obj?.shipmentItem?.quantityInvoiced - notUpdatedInvoiceItem?.quantity) <= obj?.shipmentItem?.quantity
```

Comment:
for a better visibility let's define a shipment item above:
```groovy
ShipmentItem shipmentItem = obj?.shipmentItem
```
I would also consider adding a comment above `isValid` describing in a nutshell what we treat as a valid/invalid invoice item quantity.